packages/platform-ios/src/commands/runIOS/index.js

```diff
@@ -190,6 +190,18 @@ async function runOnSimulator(xcodeProject, scheme, args: FlagsT) {
 }
 
 async function runOnDevice(selectedDevice, scheme, xcodeProject, args: FlagsT) {
+  const isIOSDeployInstalled = child_process.spawnSync(
+    'ios-deploy',
+    ['--version'],
+    {encoding: 'utf8'},
+  );
+
+  if (isIOSDeployInstalled.error) {
+    return logger.error(
+      '** INSTALLATION FAILED **\nMake sure you have ios-deploy installed globally.\n(e.g "npm install -g ios-deploy")',
```

Comment:
1. use `throw CLIError` so that the exit code is not 0.
2. change message to: 

>Failed to install the app on the device because we couldn't execute the "ios-deploy" command. Please install it from the npm registry (e.g. "npm install -g ios-deploy") and try again

Reply:
I'm doing:

```js
  const {error} = iosDeployOutput;

  if (error) {
    throw new CLIError(`
      Failed to install the app on the device. We've encountered an error in "ios-deploy" command.
      Here are more details: ${error}.
    `);
  }
```

And Flow is throwing an error saying "Cannot coerce `error` to string because  `Error` [1] should not be coerced.Flow(InferError)", do you have any ideia of how to fix this @thymikee?

Reply:
yea, just pass `error.stack`, it has message and a stack trace inside

Reply:
oh and btw, once you're done with adjustments, please post screenshots to the OP so we can see how they improved :)

Reply:
With the `error.stack` the stacktrace won't look very useful at first glance:

![image](https://user-images.githubusercontent.com/6207220/56421948-477cd500-62a5-11e9-811d-caba1b085ddb.png)

What do you mean by _the OP_?

Reply:
If you meant to pass the `error` as the second argument to `CLIError` (just got that from `runAndroid` then the error is prettified but only shows up with `--verbose`:

![image](https://user-images.githubusercontent.com/6207220/56422224-a2fb9280-62a6-11e9-908e-a3e8cf78409d.png)

Reply:
By OP I mean Original Post (the PR summary you posted).

Just to be clear, we only want to show full error message straight away if unexpected error happens, not when `ios-deploy` is not installed (there we can put error in verbose mode).

CLIError works in this funny way that it will strip all newlines. Just use regular `throw Error()` and it will work as expected :)